app/src/main/java/com/pvp/app/service/TaskAutocompleteService.kt

```diff
@@ -0,0 +1,192 @@
+package com.pvp.app.service
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.Service
+import android.content.Intent
+import android.os.IBinder
+import androidx.core.app.NotificationCompat
+import androidx.health.connect.client.records.ExerciseSessionRecord
+import com.pvp.app.api.ExerciseService
+import com.pvp.app.api.HealthConnectService
+import com.pvp.app.api.TaskService
+import com.pvp.app.api.UserService
+import com.pvp.app.model.ExerciseSessionInfo
+import com.pvp.app.model.SportTask
+import com.pvp.app.model.Task
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+import java.time.Duration
+import java.time.Instant
+import java.time.LocalDate
+import java.time.ZoneId
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class TaskAutocompleteService : Service() {
+
+    @Inject
+    lateinit var exerciseService: ExerciseService
+
+    @Inject
+    lateinit var healthConnectService: HealthConnectService
+
+    @Inject
+    lateinit var taskService: TaskService
+
+    @Inject
+    lateinit var userService: UserService
+
+    companion object {
+
+        private const val NOTIFICATION_ID = Int.MAX_VALUE
+        private const val CHANNEL_ID = "Task Autocomplete"
+    }
+
+    private fun checkTaskCompletion(
+        tasks: List<Task>,
+        exercises: List<ExerciseSessionInfo>
+    ): List<SportTask> {
+        return tasks.mapNotNull { task ->
+            val activity = (task as SportTask).activity
+            var duration = task.duration ?: Duration.ZERO
```

Comment:
In case both duration and distance are null, but we have some exercise data, this could lead the task without set duration/distance to be marked as completed, right?

I think this should be somewhat adjusted for that, if that's the case. Because for now duration/distance are mandatory when creating a task, but I really believe they should not be, since user may want to assign to themself a general task, which may take unknown time `Task#duration` and which should begin at unknown time `Task#time` - just something to do along the day.